tests/test.ts

```diff
@@ -156,25 +152,68 @@ describe("Basic remix reward deploy", function () {
   });
 
   it("Should re-set a new challenge", async function () {
-    const [owner, betatester, user, betatester2] = await ethers.getSigners();
-    
     console.log("verifier address", verifier.address)
     const setChallengeTx = await remix.connect(owner).setChallenge(verifier.address, challengeHashes, 3, tokenType, payload, hash);
     await setChallengeTx.wait()
   })
 
   it("Should refuse again an invalid challenge", async function () {   
-    const [owner, betatester, user, betatester2] = await ethers.getSigners();
-
     const invalidInput = ["0x00000000000000000000000000000000d421714eddc84195ee8f80d5379cf6f6","0x0000000000000000000000000000000042858891fcb526e577de0810598b50bc","0x000000000000000000000000000000000000000000000000000000000000002b"]
     await expect(remix.connect(betatester2).publishChallenge(proof1[0], invalidInput)).to.be.revertedWith("the provided proof isn't valid")
   });
 
   it("Should accept again a challenge", async function () {   
-    const [owner, betatester, user, betatester2] = await ethers.getSigners();
-
     const publishChallengeTx = await remix.connect(betatester2).publishChallenge(proof1[0], proof1[1])
     await publishChallengeTx.wait()
   });
-});
+})
+
+describe("Add / remove trainer", function () {
+  it("Add a trainer should fail", async function () {
+    await expect(remix.connect(trainer1).addTrainer(trainer1.address)).to.be.revertedWith('account 0x5c6b0f7bf3e7ce046039bd8fabdfd3f9f5021678 is missing role 0x0000000000000000000000000000000000000000000000000000000000000000')        
+  })
+
+  it("Add a trainer should success", async function () {
```

Comment:
There should be one more test case to check that trainer can be added by owner only not by any other address

Reply:
I think this is done in "Add a trainer should fail"

Reply:
ah yes